runtime/dotnet/azurewebapp/Startup.cs

```diff
@@ -131,17 +132,13 @@ public void ConfigureServices(IServiceCollection services)
             services.AddApplicationInsightsTelemetry();
             services.AddSingleton<ITelemetryInitializer, OperationCorrelationTelemetryInitializer>();
             services.AddSingleton<ITelemetryInitializer, TelemetryBotIdInitializer>();
-            services.AddSingleton<TelemetryLoggerMiddleware>(sp =>
-            {
-                var telemetryClient = sp.GetService<IBotTelemetryClient>();
-                return new TelemetryLoggerMiddleware(telemetryClient, logPersonalInformation: settings.Telemetry.LogPersonalInformation);
-            });
-            services.AddSingleton<TelemetryInitializerMiddleware>(sp =>
-            {
-                var httpContextAccessor = sp.GetService<IHttpContextAccessor>();
-                var telemetryLoggerMiddleware = sp.GetService<TelemetryLoggerMiddleware>();
-                return new TelemetryInitializerMiddleware(httpContextAccessor, telemetryLoggerMiddleware, settings.Telemetry.LogActivities);
-            });
+            services.AddSingleton<IBotTelemetryClient, BotTelemetryClient>();
+
+            var sp = services.BuildServiceProvider();
```

Comment:
The github actions is issuing some good advice. This is expensive to run, please remove it. Instead, in line 169 you can do s.getservice there which is a factory method and will be called after the service provider is built.

If you need to use the telemetry somehwere else, just do as we did with the bot and use the factory method instead of the instance method to do deferred resolution. Ping me if you want to chat more about this.

Reply:
Sorry, I had done this for passing in TelemetryInitializerMiddleware on line 160, but missed the fact that I had access to the service provider here (which I didn't think I did). Thanks for the catch. I'll amend now,